tests/apps/test_similarity.py

```diff
@@ -368,71 +451,230 @@ def test_mean_event(self, monkeypatch):
+        ):
+            similarity.feature_vector_events(g, [2, 4], max_count_per_mode=-1)
+
+    def test_calls_exact_for_zero_samples(self):
+        """Test if function calls the exact function for zero samples"""
+        g = nx.complete_graph(5)
+        assert similarity.feature_vector_events(
+            g, [2], 1, samples=0
+        ) == similarity.prob_event_exact(g, 2, 1)
+
+    def test_correct_vector_returned(self, monkeypatch):
+        """Test if function correctly constructs the feature vector. The ``prob_event_exact``
+         and ``prob_event_mc`` function called within ``feature_vector_events`` are
+         monkeypatched to return hard-coded outputs that depend only on the orbit."""
 
         with monkeypatch.context() as m:
-            m.setattr(sf.LocalEngine, "run", mock_eng_run)
-            similarity.prob_event_mc(graph, 6, 3, samples=1, loss=1)
-            p_func = mock_eng_run.call_args[0][0]
+            m.setattr(
+                similarity,
+                "prob_event_mc",
+                lambda _graph, photons, max_count, n_mean, samples, loss: 1.0 / photons,
+            )
+            g = nx.complete_graph(8)
+            assert similarity.feature_vector_events(g, [2, 4, 8], 1, samples=1) == [
+                0.5,
+                0.25,
+                0.125,
+            ]
 
-        eng = sf.LocalEngine(backend="gaussian")
+        with monkeypatch.context() as m:
+            m.setattr(
+                similarity,
+                "prob_event_exact",
+                lambda _graph, photons, max_count, n_mean, loss: 0.5 * (1.0 / photons),
+            )
+            g = nx.complete_graph(8)
+            assert similarity.feature_vector_events(g, [2, 4, 8], 1) == [0.25, 0.125, 0.0625]
 
-        state = eng.run(p_func).state
-        cov = state.cov()
-        disp = state.displacement()
 
-        assert np.allclose(cov, 0.5 * state.hbar * np.eye(2 * dim))
-        assert np.allclose(disp, np.zeros(dim))
+class TestFeatureVectorOrbitsSampling:
+    """Tests for the function ``strawberryfields.apps.graph.similarity.feature_vector_orbits_sampling``"""
 
+    def test_invalid_orbits_list(self):
+        """Test if function raises a ``ValueError`` when the list of orbits is empty."""
+        with pytest.raises(ValueError, match="List of orbits must have at least one orbit"):
+            similarity.feature_vector_orbits_sampling([[1, 1, 0], [1, 0, 1]], [])
 
-class TestFeatureVectorSampling:
-    """Tests for the function ``strawberryfields.apps.graph.similarity.feature_vector_sampling``"""
+    def test_bad_orbit_photon_numbers(self):
+        """Test if function raises a ``ValueError`` when input is an orbit with numbers
+        below zero."""
+        with pytest.raises(ValueError, match="Cannot request orbits with photon number below zero"):
+            similarity.feature_vector_orbits_sampling([[1, 1, 0], [1, 0, 1]], [[-1, 1]])
+
+    def test_correct_vector_returned(self, monkeypatch):
```

Comment:
This is great! These are the most important tests: are we getting the values we should from the function?